### DISCUSSION TITLE
Is there anything like Automappers AssertConfigurationIsValid?

### DISCUSSION THREAD

**Discussion by RexTremendae** [riok/mapperly#712] [Q&A] [answered]
I am currently using Mapperly in a project and personally I am a big fan. But we have discovered a problem in the team - someone changed some code and an existing mapping stopped working because a property was renamed and stopped mapping. This was not obvious and took some time to discover since it was a part of the system not used daily.

I have read the documents and tried google to see if there is anything like Automappers AssertConfigurationIsValid in Mapperly, but I didn't find it. Maybe there is a feature like this that I have missed?

If there isn't already, we could really use a way of making sure no property is accidentally overstepped. A call to some validation function or an analysis rule that gives a build warning would be two great ways to achieve this.

For us, writing unit tests for all mappings is probably not an option for us, I suspect that the team rather would migrate Automapper.

**Comment by latonz** [comment, marked as answer]
There is no `AssertConfigurationIsValid` because Mapperly emits diagnostics at compile time if a configuration has flaws. Have you tried using [strict mappings](https://mapperly.riok.app/docs/configuration/mapper/#strict-property-mappings)?

**Reply by latonz** [reply]
There is also an open issue to support one-sided strict mappings: https://github.com/riok/mapperly/issues/687

**Comment by RexTremendae** [comment]
Ah, thanks, works fine. I just had to explicitly state in `.editorconfig` that I wanted to activate each rule, just saying "default all rules to error" didn't work - which I expected since it works with the CS and IDE rules.

**Comment by stigrune** [comment]
We solved this challenge by checking the generated mapping files into the git repository. Since Mapperly generates human-readable code, it’s very easy to spot mapping bugs due to renamed properties in the pull request diff view.

**Reply by latonz** [reply]
What "bugs" are your referring to? Mapperly tries to emit all "Bugs" (invalid configurations) as compile time diagnostics (e.g. unmapped source/target member when renaming a member only on one side).
